stdlib/public/runtime/MetadataLookup.cpp

```diff
@@ -313,6 +311,56 @@ _findProtocolDescriptor(llvm::StringRef mangledName) {
   return foundProtocol;
 }
 
+#pragma Type field descriptors cache
+namespace {
+struct FieldMetadataSection {
+  const FieldDescriptor *Begin, *End;
+  const FieldDescriptor *begin() const { return Begin; }
+  const FieldDescriptor *end() const { return End; }
+};
+
+struct TypeFieldMetadataState {
+  ConcurrentMap<DescriptorCacheEntry<FieldDescriptor>> FieldCache;
+  std::vector<FieldMetadataSection> SectionsToScan;
```

Comment:
Are we allowed to use std::vector from the runtime?

Reply:
That's what the rest of the caches in this file are doing too - protocols and type metadata, is that incorrect?

Reply:
`std::vector` is okay in the runtime.